### PR TITLE
Add Jacoco 

### PR DESCRIPTION
**Pull request by alphitardian**
Add code coverage using jacoco in the project. there is some adjustment in the UI test so the jacoco can work in github actions. Maybe there will be some changes along the way if the test is not run well.

### CODE REVIEW

File: .github/workflows/codecoverage.yml [hunk 1]
```diff
@@ -4,43 +4,56 @@ name: CI
 on:
   # Triggers the workflow on push or pull request events but only for the master branch
   push:
-    branches: [ main ]
+    branches: [ main, bug-fix ]
```
Comment (elfatah) [inline .github/workflows/codecoverage.yml:7]: I think this should only the `main` branch right?

File: .github/workflows/codecoverage.yml [hunk 2]
```diff
@@ -4,43 +4,56 @@ name: CI
...
   pull_request:
-    branches: [ main ]
+    branches: [ main, bug-fix ]
```
Comment (elfatah) [inline .github/workflows/codecoverage.yml:9]: Same here

**Review by elfatah — CHANGES_REQUESTED** [review]

File: .github/workflows/codecoverage.yml (hunk 1 above)
Comment (alphitardian) [reply .github/workflows/codecoverage.yml:7]: yes it should be in main branch. but for testing purpose need to try to run it on other branch.
gonna change back to main after all problem fixed

Comment (elfatah) [reply .github/workflows/codecoverage.yml:7]: Oh okay. You should add `TODO` here then so you not forget to remove it

**Review by elfatah — APPROVED** [review]

**Review by Admokonugroho — APPROVED** [review]

**Pull request merged**